### PR TITLE
Reimplement iis dataset metadata importer to rely on InformationSpace dump

### PR DESCRIPTION
**Pull request by marekhorst**
This PR is about changing the source of imported datasets from MDStore to InfoSpace dump described in #960. 

Standalone mdstore-based dataset importer is being replaced with InfoSpace importer extension realised by a dedicated converter module. MDStore-based dataset importer code (including workflow definition, unit and integration tests) is not removed in this this PR, it should be a part of #959 task.
	
One another change caused by changing the source of datasets metadata is related to transformer module responsible for dispatching datasets either to opentrials or to dataset mining scripts: relying on `DataSetReference#resourceTypeValue` instead of `DataSetReference#resourceTypeClass` and changing expected value when picking records for opentrials processing path from `^clinicalTrial$` to `^0037$`. 	
	
It also includes `textsnippet` support committed by Ioannis into this branch (kind of out of the scope of this PR). This field provided by madis scripts is not going to be exported outside IIS. It is being used internally for data curation only.

Before merging this PR with master branch we should squash all the commits into single one.

### CODE REVIEW

File: iis-wf/iis-wf-import/src/test/resources/eu/dnetlib/iis/wf/importer/dataset/data/output/dataset.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
   "formats": [
```
[the remark this answers is not in the document]
Comment (marekhorst) [reply iis-wf/iis-wf-import/src/test/resources/eu/dnetlib/iis/wf/importer/dataset/data/output/dataset.json]: Well, I guess we could go either way but to make it human readable we could start formatting them. I will leave it for now, it is not related to changes made in this PR. We could create a dedicated task to unify formatting for all the JSONs.

**Comment by marekhorst** [comment]
I have included all the changes (apart from the JSON related ones) in a new commit and pushed it into the branch. If you resolve all the conversation topics we could carry on with mering the pull request.

File: iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceReducer.java [hunk 2]
```diff
@@ -234,17 +266,39 @@ private void handleResult(final String id, Map<String, List<QualifiedOafJsonReco
             return;
         }
         if (resultApprover.approve(oafObj)) {
+            
             DocumentMetadata docMeta = docMetaConverter.convert(oafObj.getEntity());
             if (docMeta!=null) {
-                outputs.write(outputNameDocumentMeta, new AvroKey<DocumentMetadata>(docMeta));    
+                outputs.write(outputNameDocumentMeta, new AvroKey<DocumentMetadata>(docMeta));
```
[the remark this answers is not in the document]
Comment (przemyslawjacewicz) [reply iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceReducer.java]: I think this was not changed

File: iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceReducer.java [hunk 3]
```diff
@@ -234,17 +266,39 @@ private void handleResult(final String id, Map<String, List<QualifiedOafJsonReco
...
+            }
+            
+            if (acceptAsDataset(oafObj)) {
+                DataSetReference datasetMeta = datasetMetaConverter.convert(oafObj.getEntity());
+                if (datasetMeta != null) {
+                    outputs.write(outputNameDatasetMeta, new AvroKey<DataSetReference>(datasetMeta));
```
[the remark this answers is not in the document]
Comment (przemyslawjacewicz) [reply iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceReducer.java]: I think this was not changed

File: iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/converter/DatasetMetadataConverter.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+    /**
+     * Handles persons.
+     * 
+     * @param relations person result relations
+     * @param builder
+     * @return builder with persons set
+     * @throws IOException
+     */
+    private DataSetReference.Builder handlePersons(ResultProtos.Result result, DataSetReference.Builder builder)
```
[the remark this answers is not in the document]
Comment (przemyslawjacewicz) [reply iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/converter/DatasetMetadataConverter.java]: Method still has `throws IOException` but it is not necessary anymore

File: iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/converter/MetadataConverterUtils.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            List<String> results = new ArrayList<String>(source.size());
+            for (StructuredProperty current : source) {
+                if (fieldApprover.approve(current.getDataInfo())) {
+                    results.add(current.getValue());
+                }
+            }
+            return results;
+        } else {
+            return Collections.emptyList();
+        }
```
[the remark this answers is not in the document]
Comment (przemyslawjacewicz) [reply iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/converter/MetadataConverterUtils.java]: `if` condition is not necessary I think, the whole method body can be just
```java
            return source.stream()
                    .filter(x -> fieldApprover.approve(x.getDataInfo()))
                    .map(StructuredProperty::getValue)
                    .collect(Collectors.toList());
```

**Comment by przemyslawjacewicz** [comment]
Review v2 done with two minor changes :)

**Comment by marekhorst** [comment]
Thanks, I've just pushed new commit with fixes.

**Comment by marekhorst** [comment]
Merged with https://github.com/openaire/iis/commit/f814f3973d969f7aa3212d45315079d9e279094c squashed commit.

**Pull request closed without merging**